### ISSUE TITLE
Prefect 2 import time and memory consumption is singifically higher than v1

### ISSUE THREAD

**Issue by jezekra1** [PrefectHQ/prefect#9596]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

## Importing prefect 2.10.9 takes quite long compared to version 1
**prefect v2:**
```
time python -c 'import prefect'
python -c 'import prefect'  2,26s user 1,06s system 146% cpu 2,273 total
```
**prefect v1:**
```
time python -c 'import prefect'
python -c 'import prefect'  0,89s user 0,10s system 99% cpu 0,988 total
```
It takes more than `2s` on my local machine. In our CI pipeline, it took even longer, I noticed this when some of our pipeline tests started timing out after migrating from prefect v1 to prefect v2.

## Importing prefect 2.10.9 takes quite lot of memory compared to version 1
Attaching the flamegraphs from the following:
```
memray run -o 'prefect-memory.bin' -c 'import prefect'
memray flamegraph prefect-memory.bin
```
[memray-flamegraph.zip](https://github.com/PrefectHQ/prefect/files/11495627/memray-flamegraph.zip)

**prefect v2:** ~90MB
**prefect v1:** ~63 MB

This is 42% increase in memory consumption compared to v1.

Keeping the memory consumption low is especially important for us, because we use processes to deploy flows and every new subprocess now takes ~250 MB. This is affecting the parallelism we can afford without increasing the memory requests/limits of the container. We had to decrease the number of flows we can process simultaneously.

I noticed that part of the memory usage (~17MB) `prefect-dask` which is also installed but sometimes not needed. Another big part is the server side of the library which is probably unnecessary for the agent.
 
Apart from this the experience with v2 has been a blast, thank you for the good work!
Is there something we can do to help optimize this? :)

### Reproduction

```python3
# time
time python -c 'import prefect'

# memory
memray run -o 'prefect-memory.bin' -c 'import prefect'
memray flamegraph prefect-memory.bin
```

### Versions

```Text
Tested on prefect 2.10.8 and 2.10.9 compared to prefect 1.4.1
```

### Additional context

Possibly related to: https://github.com/PrefectHQ/prefect/issues/9529

**Comment by madkinsz** [comment]
I'm actively working on this in:

- https://github.com/PrefectHQ/prefect/pull/9558
- #9557
- #9571 
- #9577

We're likely to publish a client library without the server as well in the future.

**Comment by jezekra1** [comment]
> We're likely to publish a client library without the server as well in the future.

This would be perfect for reducing the memory footprint of the agents/workers.

> We've deferred loading of collections

This is also helpful for us, we don't need dask in certain cases! :)

> Do you mind if I close this?

Sure, thanks for the update

**Comment by madkinsz** [comment]
> This would be perfect for reducing the memory footprint of the agents/workers.

After #9577 we won't load any server-side schemas but the client-side schemas will still need to be defined.

Thanks!
